code/game/objects/items/mail.dm

```diff
@@ -0,0 +1,305 @@
+
+	var/obj/junk = /obj/item/paper/fluff/junkmail_generic
+	var/special_name = FALSE
+
+	if(prob(25))
+		special_name = TRUE
+		junk = pick(list(/obj/item/paper/pamphlet/gateway, /obj/item/paper/pamphlet/centcom/visitor_info, /obj/item/paper/fluff/junkmail_redpill, /obj/effect/decal/cleanable/ash))
+
+	var/list/junk_names = list(
+		/obj/item/paper/pamphlet/gateway = "[initial(name)] for [pick(GLOB.adjectives)] adventurers",
+		/obj/item/paper/pamphlet/centcom/visitor_info = "[initial(name)] for info about visiting nanotrasen space stations",
+		/obj/item/paper/fluff/junkmail_redpill = "[initial(name)] for those feeling [pick(GLOB.adjectives)] working at Nanotrasen",
+		/obj/effect/decal/cleanable/ash = "[initial(name)] with INCREDIBLY IMPORTANT ARTIFACT- DELIVER TO SCIENCE DIVISION. HANDLE WITH CARE.",
+	)
+
+	color = pick(department_colors) //eh, who gives a shit.
+	name = special_name ? junk_names[junk] : "important [initial(name)]"
+
+	junk = new junk(src)
+	return TRUE
+
+/obj/item/mail/proc/disposal_handling(disposal_source, obj/structure/disposalholder/disposal_holder, obj/machinery/disposal/disposal_machine, hasmob)
+	if(!hasmob)
+		disposal_holder.destinationTag = sort_tag
+
+/// Subtype that's always junkmail
+/obj/item/mail/junkmail/Initialize(mapload)
+	. = ..()
+	junk_mail()
+
+/// Crate for mail from CentCom.
+/obj/structure/closet/crate/mail
+	name = "mail crate"
+	desc = "A certified post crate from CentCom."
+	icon_state = "mail"
+
+/obj/structure/closet/crate/mail/update_icon()
+	. = ..()
+	if(opened)
+		icon_state = "[initial(icon_state)]open"
+		if(locate(/obj/item/mail) in src)
+			icon_state = initial(icon_state)
+	else
+		icon_state = "[initial(icon_state)]sealed"
+
+/// Fills this mail crate with N pieces of mail, where N is the lower of the amount var passed, and the maximum capacity of this crate. If N is larger than the number of alive human players, the excess will be junkmail.
+/obj/structure/closet/crate/mail/proc/populate(amount)
+	var/mail_count = min(amount, storage_capacity)
+	// Fills the
+	var/list/mail_recipients = list()
+
+	for(var/mob/living/carbon/human in GLOB.player_list)
+		if(human.stat == DEAD || !human.mind)
+			continue
+		// Skip wizards, nuke ops, cyborgs; Centcom does not send them mail
+		if(human.faction.Find("Station", 1, 0) != 0)
+			continue
+
+		mail_recipients += human.mind
+
+	for(var/i in 1 to mail_count)
+		var/obj/item/mail/new_mail
+		if(prob(FULL_CRATE_LETTER_ODDS))
+			new_mail = new /obj/item/mail(src)
+		else
+			new_mail = new /obj/item/mail/envelope(src)
+
+		var/datum/mind/recipient = pick_n_take(mail_recipients)
+		if(recipient)
+			new_mail.initialize_for_recipient(recipient)
+		else
+			new_mail.junk_mail()
+
+	update_icon()
+
+/// Crate for mail that automatically depletes the economy subsystem's pending mail counter.
+/obj/structure/closet/crate/mail/economy/Initialize(mapload)
+	. = ..()
+	populate(SSeconomy.mail_waiting)
+	SSeconomy.mail_waiting = 0
```

Comment:
Do you intend for the mail to get reset to 0 even if the crate cannot hold the amount of waiting mail?